--- UT3AVRIL.uc	Thu Jan 15 04:12:56 1970
+++ UT3AVRIL.uc	Thu Jan 15 04:12:56 1970
@@ -108,6 +108,13 @@
 {
      LockOnSound=Sound'UT3Weapons2.Generic.LockOn'
      FireModeClass(0)=Class'UT3Style.UT3AVRiLFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponSelect"
+     PutDownAnim="WeaponPutDown"
+     SelectAnimRate=1.000000
      SelectSound=Sound'UT3Weapons.FlakCannon.FlakCannonTakeOut'
      Description="The Longbow Anti-Vehicle Rocket Launcher, a.k.a. the "
      Priority=39
@@ -120,4 +127,5 @@
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=364,Y1=213,X2=435,Y2=238)
      ItemName="UT3 Longbow AVRiL"
+     Mesh=None
 }
--- UT3AVRiLFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3AVRiLFire.uc	Thu Jan 15 04:12:56 1970
@@ -22,6 +22,9 @@
 {
      AVRiLSoundVolume=0.700000
      KickMomentum=(X=0.000000,Z=0.000000)
+     FireAnim="WeaponFire"
+     ReloadAnim="weaponreload"
+     ReloadAnimRate=0.900000
      FireSound=Sound'UT3Weapons2.AVRiL.AVRILFire'
      ReloadSound=Sound'UT3Weapons2.AVRiL.AVRILReload'
      AmmoClass=Class'UT3Style.UT3AVRiLAmmo'
--- UT3BioChargedFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3BioChargedFire.uc	Thu Jan 15 04:12:56 1970
@@ -8,6 +8,32 @@
 
 var() Sound ChargedAmbientSound;
 
+
+function ModeHoldFire()
+{
+    if ( Weapon.AmmoAmount(ThisModeNum) > 0 )
+    {
+        Super.ModeHoldFire();
+        GotoState('Hold');
+    }
+}
+
+function float MaxRange()
+{
+	return 1500;
+}
+
+simulated function bool AllowFire()
+{
+    return (Weapon.AmmoAmount(ThisModeNum) > 0 || GoopLoad > 0);
+}
+
+simulated function PlayStartHold()
+{
+    Weapon.PlayAnim('WeaponAltCharge', 0.9, 0.1);
+}
+
+
 state Hold
 {
     simulated function BeginState()
@@ -30,6 +56,7 @@
             Instigator.AmbientSound = ChargedAmbientSound;
             Instigator.SoundRadius = 50;
             Instigator.SoundVolume = 50;
+            Weapon.LoopAnim(FireLoopAnim);
         }
     }
 
@@ -68,7 +95,11 @@
 defaultproperties
 {
      ChargedAmbientSound=Sound'UT3Weapons2.BioRifle.BioRifleLoaded'
+     GoopUpRate=0.350000
      HoldSound=Sound'UT3Weapons2.BioRifle.BioRifleLoad'
+     FireAnim="WeaponAltFire"
+     FireLoopAnim="weaponaltidle"
+     FireAnimRate=2.000000
      FireSound=Sound'UT3Weapons2.BioRifle.BioRifleAltFire'
      FireRate=0.350000
      AmmoClass=Class'UT3Style.UT3BioAmmo'
--- UT3BioFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3BioFire.uc	Thu Jan 15 04:12:56 1970
@@ -6,8 +6,46 @@
 
 class UT3BioFire extends BioFire;
 
+function PlayFiring()
+{
+	if ( Weapon.Mesh != None )
+	{
+		if ( FireCount > 0 )
+		{
+			if ( Weapon.HasAnim(FireAnim) )
+			{
+				If ( Weapon.PlayAnim(FireAnim) )
+                {
+                  Weapon.PlayAnim('WeaponFire2', 1.0, 0.1);
+                }
+                Else If ( Weapon.PlayAnim('WeaponFire2') )
+                {
+                  Weapon.PlayAnim('WeaponFire3', 1.0, 0.1);
+                }
+                Else If ( Weapon.PlayAnim('WeaponFire3') )
+                {
+                  Weapon.PlayAnim(FireAnim, 1.0, 0.1);
+                }
+			}
+			else
+			{
+				Weapon.PlayAnim(FireAnim, FireAnimRate, TweenTime);
+			}
+		}
+		else
+		{
+			Weapon.PlayAnim(FireAnim, FireAnimRate, TweenTime);
+		}
+	}
+    Weapon.PlayOwnedSound(FireSound,SLOT_Interact,TransientSoundVolume,,TransientSoundRadius,Default.FireAnimRate/FireAnimRate,false);
+    ClientPlayForceFeedback(FireForce);  // jdf
+
+    FireCount++;
+}
+
 defaultproperties
 {
+     FireAnim="WeaponFire2"
      FireSound=Sound'UT3Weapons2.BioRifle.BioRifleFire'
      FireRate=0.350000
      AmmoClass=Class'UT3Style.UT3BioAmmo'
--- UT3BioGlob.uc	Thu Jan 15 04:12:56 1970
+++ UT3BioGlob.uc	Thu Jan 15 04:12:56 1970
@@ -14,4 +14,5 @@
      MyDamageType=Class'UT3Style.UT3DamTypeBioGlob'
      ImpactSound=Sound'UT3Weapons2.BioRifle.BioRifleStick'
      LifeSpan=12.000000
+     Skins(0)=None
 }
--- UT3BioRifle.uc	Thu Jan 15 04:12:56 1970
+++ UT3BioRifle.uc	Thu Jan 15 04:12:56 1970
@@ -6,10 +6,29 @@
 
 class UT3BioRifle extends BioRifle;
 
+simulated function AnimEnd(int Channel)
+{
+    local name anim;
+    local float frame, rate;
+    GetAnimParams(0, anim, frame, rate);
+
+    if (anim == 'AltFire')
+        LoopAnim('WeaponAltIdle', 1.0, 1.0);
+    else
+        Super.AnimEnd(Channel);
+}
+
 defaultproperties
 {
      FireModeClass(0)=Class'UT3Style.UT3BioFire'
      FireModeClass(1)=Class'UT3Style.UT3BioChargedFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
+     SelectAnimRate=1.000000
      SelectSound=Sound'UT3Weapons2.BioRifle.BioRifleTakeOut'
      Description="The GES BioRifle processes Tarydium from its stable crystalline form into a reactive mutagenic sludge. It can rapidly disperse these toxins for wide-area coverage, or fire a virulent payload of variable, but usually lethal, capacity. In layman's terms, this means the BioRifle can pepper an area with small globs of biosludge, or launch one noxious glob at the target. The BioRifle's ability to carpet an area with a toxic minefield makes it a notoriously effective defensive weapon."
      Priority=25
@@ -22,5 +41,7 @@
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=300,Y1=200,X2=363,Y2=229)
      ItemName="UT3 Bio Rifle"
+     Mesh=None
+     UV2Mode=UVM_LightMap
      TransientSoundVolume=0.920000
 }
--- UT3BioRiflePickup.uc	Thu Jan 15 04:12:56 1970
+++ UT3BioRiflePickup.uc	Thu Jan 15 04:12:56 1970
@@ -11,5 +11,6 @@
      InventoryType=Class'UT3Style.UT3BioRifle'
      PickupMessage="Bio Rifle"
      PickupSound=Sound'UT3PickupSounds.Generic.BioRiflePickup'
+     StaticMesh=None
      TransientSoundVolume=1.150000
 }
--- UT3Enforcer.uc	Thu Jan 15 04:12:56 1970
+++ UT3Enforcer.uc	Thu Jan 15 04:12:56 1970
@@ -137,6 +137,12 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3EnforcerFire'
      FireModeClass(1)=Class'UT3Style.UT3EnforcerAltFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdleB"
+     AimAnim="WeaponIdleB"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Weapons.Enforcer.EnforcerTakeOut'
      bShowChargingBar=False
      bUseOldWeaponMesh=False
@@ -150,4 +156,6 @@
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=299,Y1=171,X2=355,Y2=199)
      ItemName="UT3 Enforcer"
+     Mesh=None
+     DrawScale=1.200000
 }
--- UT3EnforcerAltFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3EnforcerAltFire.uc	Thu Jan 15 04:12:56 1970
@@ -46,4 +46,6 @@
 
 defaultproperties
 {
+     FireAnim="weaponfire_side"
+     FireEndAnim="weapontranition_fromside"
 }
--- UT3EnforcerFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3EnforcerFire.uc	Thu Jan 15 04:12:56 1970
@@ -12,6 +12,7 @@
      DamageMin=20
      DamageMax=20
      Momentum=1000.000000
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.Enforcer.EnforcerPrimary2'
      FireRate=0.360000
      AmmoClass=Class'UT3Style.UT3EnforcerAmmo'
--- UT3FlakAltFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3FlakAltFire.uc	Thu Jan 15 04:12:56 1970
@@ -8,6 +8,7 @@
 
 defaultproperties
 {
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.FlakCannon.FlakCannonAltFire3'
      FireRate=1.000000
      AmmoClass=Class'UT3Style.UT3FlakAmmo'
--- UT3FlakCannon.uc	Thu Jan 15 04:12:56 1970
+++ UT3FlakCannon.uc	Thu Jan 15 04:12:56 1970
@@ -10,7 +10,14 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3FlakFire'
      FireModeClass(1)=Class'UT3Style.UT3FlakAltFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Weapons.FlakCannon.FlakCannonTakeOut'
+     OldPlayerViewOffset=(Y=7.000000)
      Description="Trident Defensive Technologies continues to tweak and refine the flak cannon with their newly released Mk4 "
      Priority=52
      HudColor=(B=128,G=255)
@@ -19,7 +26,9 @@
      CustomCrossHairScale=1.500000
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairFlakCannon"
      PickupClass=Class'UT3Style.UT3FlakCannonPickup'
+     PlayerViewPivot=(Yaw=0)
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=65,Y1=214,X2=131,Y2=240)
      ItemName="UT3 Flak Cannon"
+     Mesh=None
 }
--- UT3FlakCannonPickup.uc	Thu Jan 15 04:12:56 1970
+++ UT3FlakCannonPickup.uc	Thu Jan 15 04:12:56 1970
@@ -11,5 +11,7 @@
      InventoryType=Class'UT3Style.UT3FlakCannon'
      PickupMessage="Flak Cannon"
      PickupSound=Sound'UT3PickupSounds.Generic.FlakCannonPickup'
+     StaticMesh=None
+     UV2Mode=UVM_LightMap
      TransientSoundVolume=1.000000
 }
--- UT3FlakFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3FlakFire.uc	Thu Jan 15 04:12:56 1970
@@ -8,6 +8,7 @@
 
 defaultproperties
 {
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.FlakCannon.FlakCannonPrimary2'
      FireRate=1.100000
      AmmoClass=Class'UT3Style.UT3FlakAmmo'
--- UT3ImpactHammer.uc	Thu Jan 15 04:12:56 1970
+++ UT3ImpactHammer.uc	Thu Jan 15 04:12:56 1970
@@ -41,11 +41,11 @@
 {
     if (FireMode[0].bIsFiring)
     {
-        LoopAnim('Charged');
+        LoopAnim('WeaponChargedIdle');
     }
     else if (FireMode[1].bIsFiring)
     {
-        LoopAnim('Charged');
+        LoopAnim('WeaponChargedIdle');
     }
 }
 
@@ -134,15 +134,20 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3ImpactHammerFire'
      FireModeClass(1)=Class'UT3Style.UT3ImpactHammerEMPFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponSelect"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Weapons.ImpactHammer.ImpactHammerTakeOut'
-     Description="The Impact Hammer, originally designed for sub-surface drift mining, can focus several hundred metric tons of pressure into the space of a few square centimeters. The impact, and resulting shock wave, easily destabilizes solid stone to speed ore separation and excavation. After several mining accidents highlighted the Hammer's devastating effect on the human body, the N.E.G. military took interest. To improve the Hammer's effectiveness against vehicles with shock-absorbing armor, an alternate electromagnetic pulse (EMP) mode was added. The EMP violently scrambles onboard computer systems, quickly leading to engine failure. Field testing showed the EMP has a similar effect on infantry powerups, knocking them off soldiers and enabling battlefield recovery."
+     Description="Originally designed for sub-surface drift mining, this weapon can focus several hundred metric tons of pressure into the space of a few square centimeters. The impact, and resulting shock wave, easily destabilizes solid stone to speed ore separation and excavation. After several mining accidents highlighted the Hammer's devastating effect on the human body, the N.E.G. military took interest.|To improve the Hammer's effectiveness against vehicles with shock-absorbing armor, an alternate electromagnetic pulse (EMP) mode was added. The EMP violently scrambles onboard comptuer systems, quickly leading to engine failure. Field testing showed the EMP has a similar effect on infantry powerups, knocking them off soldiers and enabling battlefield recovery."
      Priority=10
-     HudColor=(B=128,G=255)
      CustomCrosshair=22
      CustomCrossHairColor=(B=128,G=255)
      CustomCrossHairScale=1.500000
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairImpactHammer"
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
-     IconCoords=(X1=226,Y1=162,X2=294,Y2=191)
      ItemName="UT3 Impact Hammer"
+     Mesh=None
 }
--- UT3ImpactHammerEMPFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3ImpactHammerEMPFire.uc	Thu Jan 15 04:12:56 1970
@@ -17,6 +17,8 @@
      SelfDamageScale=0.000000
      MinSelfDamage=0.000000
      ChargingSound=Sound'UT3Weapons2.ImpactHammer.ImpactHammerLoop'
+     PreFireAnim="WeaponCharge"
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.ImpactHammer.ImpactHammerEMP'
      FireRate=1.100000
 }
--- UT3ImpactHammerFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3ImpactHammerFire.uc	Thu Jan 15 04:12:56 1970
@@ -14,6 +14,8 @@
      MaxDamage=140.000000
      SelfForceScale=1.200000
      ChargingSound=Sound'UT3Weapons.ImpactHammer.ImpactHammerLoop'
+     PreFireAnim="WeaponCharge"
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.ImpactHammer.ImpactHammerFire'
      FireRate=1.100000
 }
--- UT3InstagibRifle.uc	Thu Jan 15 04:12:56 1970
+++ UT3InstagibRifle.uc	Thu Jan 15 04:12:56 1970
@@ -54,8 +54,15 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3SuperShockBeamFire'
      FireModeClass(1)=Class'UT3Style.UT3SuperShockBeamFire'
+     IdleAnim="Idle"
+     RestAnim="rest"
+     AimAnim="Aim"
+     RunAnim="Run"
+     SelectAnim="Pickup"
+     PutDownAnim="PutDown"
      AIRating=1.000000
      bCanThrow=False
+     OldPlayerViewPivot=(Yaw=49152)
      Priority=7
      CustomCrosshair=30
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairShockRifle"
--- UT3LinkGun.uc	Thu Jan 15 04:12:56 1970
+++ UT3LinkGun.uc	Thu Jan 15 04:12:56 1970
@@ -24,8 +24,15 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3LinkGunFire'
      FireModeClass(1)=Class'UT3Style.UT3LinkGunAltFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Style.LinkGun.LinkGunRaise'
      AIRating=0.710000
+     OldPlayerViewPivot=(Yaw=49152)
      Priority=31
      HudColor=(B=0,R=255)
      CustomCrosshair=24
@@ -33,8 +40,10 @@
      CustomCrossHairScale=1.300000
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairLinkGun"
      PickupClass=Class'UT3Style.UT3LinkGunPickup'
+     PlayerViewPivot=(Yaw=49152)
      AttachmentClass=Class'UT3Style.UT3LinkGunAttachment'
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=226,Y1=233,X2=300,Y2=254)
      ItemName="UT3 Link Gun"
+     Mesh=None
 }
--- UT3LinkGunAltFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3LinkGunAltFire.uc	Thu Jan 15 04:12:56 1970
@@ -6,6 +6,30 @@
 
 class UT3LinkGunAltFire extends LinkFire;
 
+
+function PlayFiring()
+{
+	if (LinkGun(Weapon).Links <= 0 && Weapon.AmmoAmount(ThisModeNum) >= AmmoPerFire)
+		ClientPlayForceFeedback("BLinkGunBeam1");
+    Super.PlayFiring();
+}
+
+function StopFiring()
+{
+	Instigator.AmbientSound = None;
+	Instigator.SoundVolume = Instigator.Default.SoundVolume;
+    if (Beam != None)
+    {
+        Beam.Destroy();
+        Beam = None;
+    }
+    SetLinkTo(None);
+    bStartFire = true;
+    bFeedbackDeath = false;
+    if (LinkGun(Weapon).Links <= 0)
+		StopForceFeedback("BLinkGunBeam1");
+}
+
 defaultproperties
 {
      MakeLinkSound=Sound'UT3Style.LinkGun.BaseLinkGunAltFireStart'
@@ -17,6 +41,11 @@
      BeamSounds(1)=Sound'UT3Style.LinkGun.LinkGunAltFireImpactLoop'
      BeamSounds(2)=Sound'UT3Style.LinkGun.LinkGunAltFireImpactFleshLoop'
      BeamSounds(3)=Sound'UT3Style.LinkGun.LinkGunAltFireImpactFleshLoop'
+     PreFireAnim="WeaponAltFireStart"
+     FireAnim=
+     FireLoopAnim="WeaponAltFire"
+     FireEndAnim="WeaponAltFireEnd"
+     FireLoopAnimRate=0.500000
      FireRate=0.100000
      AmmoClass=Class'UT3Style.UT3LinkAmmo'
 }
--- UT3LinkGunFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3LinkGunFire.uc	Thu Jan 15 04:12:56 1970
@@ -9,6 +9,7 @@
 defaultproperties
 {
      LinkedFireSound=Sound'UT3Style.LinkGun.LinkGunFire'
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Style.LinkGun.LinkGunFire'
      FireRate=0.160000
      AmmoClass=Class'UT3Style.UT3LinkAmmo'
--- UT3LinkProjectile.uc	Thu Jan 15 04:12:56 1970
+++ UT3LinkProjectile.uc	Thu Jan 15 04:12:56 1970
@@ -25,5 +25,6 @@
      MaxSpeed=5000.000000
      Damage=26.000000
      MyDamageType=Class'UT3Style.DamTypeUT3LinkProj'
+     DrawScale=0.400000
      TransientSoundVolume=0.500000
 }
--- UT3Minigun2v.uc	Thu Jan 15 04:12:56 1970
+++ UT3Minigun2v.uc	Thu Jan 15 04:12:56 1970
@@ -171,13 +171,18 @@
 
 defaultproperties
 {
-     AttachLoc=(X=-77.000000,Y=6.000000,Z=4.000000)
      AttachRot=(Pitch=22000,Yaw=-16384)
      GearRatio=-2.370000
      Blend=1.000000
      FireModeClass(0)=Class'UT3Style.UT3MinigunFire'
      FireModeClass(1)=Class'UT3Style.UT3MinigunAltFire'
-     PutDownAnim="PutDown"
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaoponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
+     RestAnimRate=0.000000
      SelectSound=Sound'UT3Weapons2.Stinger.StingerTakeOut'
      SelectForce="SwitchToMiniGun"
      AIRating=0.100000
@@ -210,7 +215,6 @@
      LightBrightness=255.000000
      LightRadius=5.000000
      LightPeriod=3
-     Mesh=SkeletalMesh'Weapons.Minigun_1st'
      DrawScale=0.400000
      SoundRadius=400.000000
      HighDetailOverlay=Combiner'UT2004Weapons.WeaponSpecMap2'
--- UT3MinigunAltFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3MinigunAltFire.uc	Thu Jan 15 04:12:56 1970
@@ -311,7 +311,7 @@
 defaultproperties
 {
      BarrelRotationsPerSec=0.975000
-     RoundsPerRotation=4
+     RoundsPerRotation=3
      WindingSound=Sound'UT3Weapons2.Stinger.StingerRapid'
      WindUpTime=0.270000
      FiringForce="minifireb"
@@ -319,7 +319,10 @@
      ProjSpawnOffset=(X=25.000000,Y=5.000000,Z=-6.000000)
      bPawnRapidFireAnim=True
      PreFireTime=0.270000
-     FireLoopAnimRate=9.000000
+     FireAnim="WeaponFire-Secondary"
+     FireLoopAnim="WeaponFire-Secondary"
+     FireEndAnim="WeaponFireEnd"
+     FireLoopAnimRate=1.800000
      FireSound=Sound'UT3Weapons2.Stinger.StingerAltStop'
      AmmoClass=Class'UT3Style.UT3MinigunAmmo'
      AmmoPerFire=2
--- UT3MinigunFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3MinigunFire.uc	Thu Jan 15 04:12:56 1970
@@ -277,7 +277,12 @@
      Momentum=0.000000
      bPawnRapidFireAnim=True
      PreFireTime=0.270000
-     FireLoopAnimRate=9.000000
+     PreFireAnim="WeaponFireStart"
+     FireAnim="WeaponFire"
+     FireLoopAnim="WeaponFire"
+     FireEndAnim="WeaponFireEnd"
+     FireLoopAnimRate=1.200000
+     FireForce="minifireb"
      AmmoClass=Class'UT3Style.UT3MinigunAmmo'
      AmmoPerFire=1
      ShakeRotMag=(X=50.000000,Y=50.000000,Z=50.000000)
--- UT3Redeemer.uc	Thu Jan 15 04:12:56 1970
+++ UT3Redeemer.uc	Thu Jan 15 04:12:56 1970
@@ -10,6 +10,13 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3RedeemerFire'
      FireModeClass(1)=Class'UT3Style.UT3RedeemerGuidedFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
+     SelectAnimRate=1.000000
      SelectSound=Sound'UT3Weapons2.Redeemer.RedeemerTakeOut'
      Description="'Even your least effective soldiers will earn a respectable bodycount with this tactical nuclear device.' So said the original brochure for the Redeemer, still unchallenged in its role as the most powerful man-portable weapon system known through the galaxy.|The slow-moving but utterly devastating missile, affectionately known as 'Lola' by veteran soldiers, now uses an Enhanced Radiation payload. This ensures maximum tissue failure without undue property damage, perfect for modern assault-and-capture tactics.|The Redeemer's alternate mode fires the missile using the disposable fly-by-wire guidance system, though the manual recommends using this mode only in areas of relative safety."
      Priority=56
@@ -21,5 +28,6 @@
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=227,Y1=191,X2=299,Y2=233)
      ItemName="UT3 Redeemer"
+     Mesh=None
      TransientSoundVolume=0.450000
 }
--- UT3RedeemerFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3RedeemerFire.uc	Thu Jan 15 04:12:56 1970
@@ -23,6 +23,7 @@
 
 defaultproperties
 {
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons2.Redeemer.RedeemerFire'
      ProjectileClass=Class'UT3Style.UT3RedeemerProjectile'
 }
--- UT3RedeemerGuidedFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3RedeemerGuidedFire.uc	Thu Jan 15 04:12:56 1970
@@ -46,6 +46,7 @@
 
 defaultproperties
 {
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons2.Redeemer.RedeemerFire'
      ProjectileClass=Class'UT3Style.UT3RedeemerProjectile'
 }
--- UT3RocketFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3RocketFire.uc	Thu Jan 15 04:12:56 1970
@@ -9,6 +9,7 @@
 defaultproperties
 {
      TransientSoundVolume=0.400000
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons2.RocketLauncher.RocketLauncherFire'
      FireRate=1.050000
      AmmoClass=Class'UT3Style.UT3RocketAmmo'
--- UT3RocketLauncher.uc	Thu Jan 15 04:12:56 1970
+++ UT3RocketLauncher.uc	Thu Jan 15 04:12:56 1970
@@ -119,6 +119,20 @@
         return Rocket;
     }
 }
+simulated function RotateBarrel()
+{
+//FAIL.
+}
+
+simulated function UpdateBarrel(float dt)
+{
+//STOP. HAMMER TIME.
+}
+
+simulated function Plunge()
+{
+//LOL NOPE.
+}
 
 
 simulated state AnimateLoad
@@ -151,6 +165,13 @@
      LoadEndSound=Sound'UT3Weapons2.RocketLauncher.RocketLauncherLoad2'
      FireModeClass(0)=Class'UT3Style.UT3RocketFire'
      FireModeClass(1)=Class'UT3Style.UT3RocketMultiFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
+     IdleAnimRate=1.000000
      SelectSound=Sound'UT3Weapons2.RocketLauncher.RocketLauncherTakeOut'
      Description="Each year, more accidental deaths are caused by the Trident Tri-barrel Rocket Launcher than in vehicular accidents and extreme sports combined. The kill radius for its standard dumbfire rocket is surprisingly high by design - so accidentally firing upon a nearby wall, or a nearby enemy soldier, can be quite fatal for the operator. The alternate fire adds to this suicidal lethality by loading and firing up to three rockets at once, in a spread, tight spiral, or lobbed like grenades. Regardless of the grim statistics, veteran soldiers still consider the 'old 8 ball' the most expedient way to put explosive ordnance on target."
      Priority=48
@@ -161,5 +182,6 @@
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=65,Y1=189,X2=130,Y2=214)
      ItemName="UT3 Rocket Launcher"
+     Mesh=None
      TransientSoundVolume=0.600000
 }
--- UT3RocketMultiFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3RocketMultiFire.uc	Thu Jan 15 04:12:56 1970
@@ -11,12 +11,30 @@
     UT3RocketLauncher(Weapon).PlayOwnedSound(Sound'UT3Weapons2.RocketLauncher.RocketLauncherLoad1', SLOT_None,,,,,false);
 }
 
+function PlayFiring()
+{
+    if (Load == 1.0)
+        FireAnim = 'WeaponAltFireLaunch1';
+    else if (Load == 2.0)
+        FireAnim = 'WeaponAltFireLaunch2';
+    else if (Load == 3.0)
+        FireAnim = 'WeaponAltFireLaunch3';
+    Super.PlayFiring();
+    RocketLauncher(Weapon).PlayFiring((Load == MaxLoad));
+	Weapon.OutOfAmmo();
+}
+
+simulated function Plunge()
+{
+    if (Load == 1.0)
+        FireAnim = 'WeaponAltFireQueue1';
+    else if (Load == 2.0)
+        FireAnim = 'WeaponAltFireQueue2';
+    else if (Load == 3.0)
+        FireAnim = 'WeaponAltFireQueue3';
+}
+
 defaultproperties
 {
-     MaxHoldTime=2.600000
-     TransientSoundVolume=0.400000
-     FireSound=Sound'UT3Weapons2.RocketLauncher.RocketLauncherFire'
-     FireRate=1.050000
-     AmmoClass=Class'UT3Style.UT3RocketAmmo'
-     ProjectileClass=Class'UT3Style.UT3RocketProj'
+     FireAnim=
 }
--- UT3ShockAltFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3ShockAltFire.uc	Thu Jan 15 04:12:56 1970
@@ -19,6 +19,7 @@
 defaultproperties
 {
      TransientSoundVolume=0.800000
+     FireAnim="WeaponAltFire"
      FireSound=Sound'UT3Weapons.ShockRifle.ShockRifleAlt'
      AmmoClass=Class'UT3Style.UT3ShockAmmo'
      ProjectileClass=Class'UT3Style.UT3ShockBall'
--- UT3ShockFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3ShockFire.uc	Thu Jan 15 04:12:56 1970
@@ -40,6 +40,7 @@
 {
      DamageType=Class'UT3Style.DamTypeUT3ShockBeam'
      TransientSoundVolume=0.470000
+     FireAnim="WeaponFire"
      FireSound=Sound'UT3Weapons.ShockRifle.ShockRiflePrimary'
      FireRate=0.770000
      AmmoClass=Class'UT3Style.UT3ShockAmmo'
--- UT3ShockRifle.uc	Thu Jan 15 04:12:56 1970
+++ UT3ShockRifle.uc	Thu Jan 15 04:12:56 1970
@@ -10,9 +10,16 @@
 {
      FireModeClass(0)=Class'UT3Style.UT3ShockFire'
      FireModeClass(1)=Class'UT3Style.UT3ShockAltFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Weapons.ShockRifle.ShockRifleTakeOut'
      AIRating=0.650000
      bSniping=True
+     OldPlayerViewPivot=(Pitch=0,Yaw=-5000,Roll=0)
      Description="The ASMD Shock Rifle has changed little since its first incorporation into the Liandri Tournament. The ASMD sports two firing modes: a thin photon beam, and a sphere of unstable plasma energy. These modes are each deadly in their own right, but used together they can neutralize opponents in a devastating shockwave. The energy spheres can be detonated with the photon beam, releasing the explosive energy of the anti-photons in the plasma's EM containment field."
      Priority=22
      HudColor=(R=160)
@@ -21,8 +28,10 @@
      CustomCrossHairScale=1.000000
      CustomCrossHairTextureName="Crosshairs.HUD.Crosshair_Cross2"
      PickupClass=Class'UT3Style.UT3ShockRiflePickup'
+     PlayerViewPivot=(Pitch=0,Yaw=49152)
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=363,Y1=190,X2=445,Y2=213)
      ItemName="UT3 Shock Rifle"
+     Mesh=None
      TransientSoundVolume=1.000000
 }
--- UT3SniperFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3SniperFire.uc	Thu Jan 15 04:12:56 1970
@@ -1,18 +1,9 @@
 //=============================================================================
-// UT3SniperFire.uc
-// Head Shot.
-// 2008, GreatEmerald
+// UT3SniperFire.
 //=============================================================================
-
-class UT3SniperFire extends ClassicSniperFire;
+class UT3SniperFire extends SniperFire;
 
 defaultproperties
 {
-     DamageTypeHeadShot=Class'UT3Style.DamTypeUT3HeadShot'
-     DamageType=Class'UT3Style.DamTypeUT3Sniper'
-     DamageMin=70
-     DamageMax=70
-     FireSound=Sound'UT3Weapons2.Sniper.SniperShot'
-     AmmoClass=Class'UT3Style.UT3SniperAmmo'
-     aimerror=600.000000
+     FireAnim="WeaponFire"
 }
--- UT3SniperRifle.uc	Thu Jan 15 04:12:56 1970
+++ UT3SniperRifle.uc	Thu Jan 15 04:12:56 1970
@@ -131,9 +131,16 @@
 defaultproperties
 {
      FireModeClass(0)=Class'UT3Style.UT3SniperFire'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
+     PutDownAnim="WeaponPutDown"
      SelectSound=Sound'UT3Weapons2.Sniper.SniperTakeOut'
      AIRating=0.700000
      CurrentRating=0.700000
+     OldPlayerViewPivot=(Yaw=49152)
      Description="The venerable Axon Research long-range target interdiction rifle is the weapon of choice for the discerning sniper. Acquisition of a target at distance requires a steady hand, but the rewards merit the effort: the high-caliber round is lethal at any range. With a precise headshot, the target will be neutralized by the super-sonic bullet long before they hear the report. As they say in the N.E.G. Marines: 'any shot you hear is nothing to be worried about.'"
      Priority=20
      HudColor=(B=64,G=0,R=255)
@@ -141,9 +148,12 @@
      CustomCrossHairColor=(B=64,G=0,R=255)
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairSniperRifle"
      PickupClass=Class'UT3Style.UT3SniperRiflePickup'
+     PlayerViewPivot=(Yaw=0)
      AttachmentClass=Class'UT3Style.UT3SniperAttachment'
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=362,Y1=260,X2=445,Y2=286)
      ItemName="UT3 Sniper Rifle"
+     Mesh=None
+     DrawScale=2.000000
      TransientSoundVolume=0.730000
 }
--- UT3SuperShockBeamFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3SuperShockBeamFire.uc	Thu Jan 15 04:12:56 1970
@@ -78,6 +78,7 @@
      DamageMax=1000
      Momentum=100000.000000
      TransientSoundVolume=0.700000
+     FireAnim="WeaponFireInstiGib"
      FireSound=Sound'UT3Weapons2.Generic.InstagibRifleFire'
      FireRate=1.100000
      AmmoClass=Class'XWeapons.SuperShockAmmo'
--- UT3Translocator.uc	Thu Jan 15 04:12:56 1970
+++ UT3Translocator.uc	Thu Jan 15 04:12:56 1970
@@ -36,15 +36,23 @@
      AmmoChargeRate=0.800000
      FireModeClass(0)=Class'UT3Style.UT3TranslocatorFire'
      FireModeClass(1)=Class'UT3Style.UT3TranslocatorActivate'
+     IdleAnim="WeaponIdle"
+     RestAnim="WeaponIdle"
+     AimAnim="WeaponIdle"
+     RunAnim="WeaponIdle"
+     SelectAnim="WeaponEquip"
      SelectSound=Sound'UT3Style.Translocator.TranslocatorRaise'
      Description="The Translocator was originally designed by Liandri R&D for rapid rescue of expensive mining equipment during tunnel collapses and related emergencies. The technology also saved couintless lives, but not without a cost: rapid deresolution and reconstitution led to synaptic disruptions, and the debilitating symptoms like Teleportation Related Dementia (TReDs).||Today, after years of lucrative military development contracts, portable teleportation technology has been declared 'sufficiently safe' for regular use by frontline infantry."
+     Priority=0
      HudColor=(B=128,G=255,R=255)
+     CustomCrosshair=19
      CustomCrossHairColor=(B=128,G=255,R=255)
      CustomCrossHairScale=1.200000
      CustomCrossHairTextureName="UT3HUD.Crosshairs.UT3CrosshairDefault"
      IconMaterial=TexScaler'UT3HUD.Icons.UT3IconsScaled'
      IconCoords=(X1=300,Y1=230,X2=361,Y2=256)
      ItemName="UT3 Translocator"
+     Mesh=None
      TransientSoundVolume=0.700000
      TransientSoundRadius=1000.000000
 }
--- UT3TranslocatorActivate.uc	Thu Jan 15 04:12:56 1970
+++ UT3TranslocatorActivate.uc	Thu Jan 15 04:12:56 1970
@@ -140,4 +140,5 @@
 {
      TransEffects(0)=(Overlay=Shader'XGameShaders.PlayerShaders.PlayerTransRed',Effect=Class'UT3Style.UT3TranslocatorEffectRed')
      TransEffects(1)=(Overlay=Shader'XGameShaders.PlayerShaders.PlayerTrans',Effect=Class'UT3Style.UT3TranslocatorEffectBlue')
+     FireAnim="weaponreturn"
 }
--- UT3TranslocatorFire.uc	Thu Jan 15 04:12:56 1970
+++ UT3TranslocatorFire.uc	Thu Jan 15 04:12:56 1970
@@ -61,5 +61,6 @@
      RecallFailedSound=Sound'WeaponSounds.BaseGunTech.BSeekLost1'
      TransFireSound=ProceduralSound'UT3Style.Translocator.TranslocatorFire'
      RecallFireSound=SoundGroup'UT3Style.Translocator.TranslocatorRecall'
+     FireAnim="WeaponFire"
      ProjectileClass=Class'UT3Style.UT3TranslocatorDiskRed'
 }
